### DISCUSSION TITLE
Toggle speed shortcut

### DISCUSSION THREAD

**Discussion by robsonhermes** [diasurgical/DevilutionX#7399] [Ideas]
Hello

First, thanks for devilutionX! 
A suggestion for a QOL improvement.

**Idea**: A shortcut to toggle speed max/regular value

**Context**: I find it quite useful to bump speed to the max when I go back to town, or when I hit a dead end in a dungeon and need to go all the way back. Then I switch back to what I consider a regular gaming speed when I reach unexplored areas. But it's a bit annoying to every time hit ESC, and adjust the speed. Would be awesome to toggle quickly.

**How it could work**: 
In the keyboard shortcut options, we could create a new option "Toggle speed". Perhaps every time the player manually adjusts the speed, we could store in a variable that's the player's desired "regular gaming speed". When pushing the defined shortcut, the game would toggle between MAX speed, and the stored value. If the player manually sets the speed to the MAX (so this would be stored in the previously mentioned variable), then it would be basically a no-op, switching between MAX and MAX.

Hopefully that makes sense.

**Comment by AJenbo** [comment]
esc, down, enter, up, up, enter, esc

**Reply by robsonhermes** [reply]
I think a hotkey would be still easier (more practical) and the steps you mentioned allow to toggle between min and max. The suggestion above would help players who play are a speed higher than the min.

**Comment by Arratai** [comment]
I had a similar idea, and even though I'm nowhere near experienced enough to put it directly into devilution, I made a very simple python script that toggles between max and min speed. you can find it here: https://github.com/Arratai/Diablo-devilution-speed-toggle

its my first time doing something like this

**Reply by robsonhermes** [reply]
I knew I wasn't the only one 😆

**Comment by AJenbo** [comment]
We should make sure to add enough lua binding to allow a simple script for this.

**Comment by RandomAccess42** [comment]
Try this patch .. its very simple since there is global variable representing tick delay.. and when you modify that the speed will change. This will be reset to some default value after restart of diablo.

I use it to speed up stuff in village and when I have to walk around long distances of empty dungeons.. no cheating! lol

[0001-diablo-add-hotkey-for-changing-the-speed-j-slower-k-.txt](https://github.com/user-attachments/files/21865606/0001-diablo-add-hotkey-for-changing-the-speed-j-slower-k-.txt)
